### ISSUE TITLE
Validate not working properly

### ISSUE THREAD

**Issue by bschveitzer** [RobinBuschmann/sequelize-typescript#254]
When calling instance .validate() it returns me a sequelize object instead of ValidationError object.

Is that supposed to happen ?

**Comment by RobinBuschmann** [comment]
Hey @bschveitzer. Can you post a code example and your sequelize version please? Thank you :)

**Comment by Cyberuben** [comment]
Hi Robin,

as I see this issue has gone silent, and I'm experiencing the same issue, I'll post the code here:

```javascript
let user = new User(data);
let errors = await user.validate();
console.log(errors);
```

this will output an instance of the user that I just created, no validation errors.

I'm using VSCode and the intellisense shows the following:

![Intellisense](http://scrnsht.me/u/mGc/raw)

There is in fact an error in the data I'm supplying, as it is thrown when I'm trying to save the model, but something here seems to go wrong.

**Comment by Cyberuben** [comment]
To elaborate on the issue: I found that this problem occurs whenever you supply a validator function that returns either `true` or `false`

With the following code:

```javascript
// within my model definition
	@Unique
	@IsEmail
	@Is("emailAvailable", emailAvailable)
	@Column
	email: string;
```

**works** with the following validator:

```javascript
const emailAvailable = async function (email: string): Promise<void> {
	const user = await User.findOne({ where: { email } });

	if (user != null && user.id !== this.id) {
		throw new Error("Email already in use");
	}
};
```

**fails** with the following validator:

```javascript
const emailAvailable = async function (email: string): Promise<boolean> {
	const user = await User.findOne({ where: { email } });

	return user == null || user.id === this.id;
};
```

I'm not sure if this is intended behavior, as custom validators for Sequelize seem to be badly documented, but at least I found the source for my issues.

**Comment by RobinBuschmann** [comment]
Ah ok, I got it. Validation works differently for the two latest sequelize major versions. While `validate()` in version 3 resolves with an error if one exists, it rejects the promise  in version 4:

```ts
user.validate().then(err => ...) // 3.X
user.validate().catch(err => ...) // 4.X

```

sequelize-typescript currently supports both major versions. So the actual issue is in the sequelize-typescript typings. It should be something like
`validate(...): Promise<ValidationError /*v3*/ | Model>` than the current type definition.

**Comment by RobinBuschmann** [comment]
Is fixed with `sequelize-typescript@1.0.0-alpha.8`
